main.tf

```diff
@@ -88,6 +88,8 @@ resource "ibm_resource_key" "resource_key" {
 locals {
   cos_instance_id          = var.create_cos_instance ? ibm_resource_instance.cos_instance[0].id : var.existing_cos_instance_id
   cos_instance_guid        = var.create_cos_instance ? ibm_resource_instance.cos_instance[0].guid : element(split(":", var.existing_cos_instance_id), length(split(":", var.existing_cos_instance_id)) - 3)
+  cos_instance_name        = var.create_cos_instance ? ibm_resource_instance.cos_instance[0].name : null
+  cos_instance_crn         = var.create_cos_instance ? ibm_resource_instance.cos_instance[0].target_crn : null
```

Comment:
Is `target_crn` the same as `crn` ? Did you test it works? Perhaps also add it to the output of one of the examples so we can test its value exists. 
I'm asking because I see in the SCC module, we have:
```
output "crn" {
  description = "The CRN of the SCC instance created by this module"
  value       = resource.ibm_resource_instance.scc_instance.crn
}
```
And from working with this module, I see this works:
```
base_security_scc = [
  {
    "crn" = "crn:v1:bluemix:public:compliance:us-south:a/abac0df06b644a9cabc6e44f55b3880e:334f6dcc-2d10-4b4e-8050-95e94e1a67d7::"
    "guid" = "334f6dcc-2d10-4b4e-8050-95e94e1a67d7"
    "id" = "crn:v1:bluemix:public:compliance:us-south:a/abac0df06b644a9cabc6e44f55b3880e:334f6dcc-2d10-4b4e-8050-95e94e1a67d7::"
    "location" = "us-south"
    "name" = "base-security-services-scc"
    "plan" = "security-compliance-center-standard-plan"
  },
]
```

Reply:
I do not see any attribute named `crn` in the `ibm_resource_instance` resource. See: https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/resource_instance

But it looks like it's not documented. Let me add `crn` and consume that in one of the examples.